kubecargoload.py

```diff
@@ -43,13 +44,19 @@ def _fetch_pod_memory_usage(self):
         top_pods_output = self._execute_kubectl_top_pods()
         for line in top_pods_output.splitlines():
             columns = line.strip().split()
-            if len(columns) > 3:
-                namespace, name, _, memory_usage_pretty = columns
+            if self._cpu:
+                if len(columns) > 3:
+                    namespace, name, usage_pretty, _ = columns
+                else:
+                    name, usage_pretty, _ = columns
+                    namespace = self._namespace
+            elif len(columns) > 3:
+                namespace, name, _, usage_pretty = columns
             else:
-                name, _, memory_usage_pretty = columns
+                name, _, usage_pretty = columns
                 namespace = self._namespace
 
-            memory_usage = self._parse_quantity(memory_usage_pretty)
+            memory_usage = self._parse_quantity(usage_pretty)
 
             pod_key = (namespace, name)
             self._pod_memory_usage_data[pod_key] = memory_usage
```

Comment:
I would rewrite this part rather as this:
```py
            columns = line.strip().split()
            if len(columns) > 3:
                namespace, name, cpu_usage_pretty, memory_usage_pretty = columns
            else:
                name, cpu_usage_pretty, memory_usage_pretty = columns
                namespace = self._namespace

            usage_pretty = cpu_usage_pretty if self._cpu else memory_usage_pretty
            usage = self._parse_quantity(usage_pretty)

            pod_key = (namespace, name)
            self._pod_memory_usage_data[pod_key] = usage
```
This way it has less nested conditions and so is more readable, in my opinion.

Reply:
Sounds good. Will make the changes.